gobblin-compaction/src/main/java/gobblin/compaction/mapreduce/MRCompactor.java

```diff
@@ -788,6 +792,13 @@ protected void afterExecute(Runnable r, Throwable t) {
             // Set the dataset status to COMPACTION_COMPLETE if compaction is successful.
             jobRunner.getDataset().setState(COMPACTION_COMPLETE);
           }
+          if (MRCompactor.this.compactorListener.isPresent()) {
```

Comment:
There is a new class `SerialCompactorListener` that takes in a `List` of `CompactorListener`s and executes them one by one. This class is used by `ReflectionCompactorListenerFactory`.